src/main/scala/mirroring/Runner.scala

```diff
@@ -51,30 +51,29 @@ object Runner extends LogSupport {
     val jdbcContext                                  = config.getJdbcContext
     val writerContext: WriterContext                 = config.getWriterContext
     var query: String                                = config.query
-    var changeTrackingHandler: ChangeTrackingHandler = new ChangeTrackingHandler(config)
+    val changeTrackingHandler: ChangeTrackingHandler = new ChangeTrackingHandler(config)
 
     if (config.isChangeTrackingEnabled) {
-      if (config.CTChangesQuery.nonEmpty) {
-        changeTrackingHandler = new CustomChangeTrackingHandler(config)
-      }
       query = changeTrackingHandler.query
       writerContext.ctCurrentVersion = changeTrackingHandler.ctCurrentVersion
       jdbcContext._ctCurrentVersion = Some(changeTrackingHandler.ctCurrentVersion)
-      jdbcContext._changeTrackingLastVersion = Some(changeTrackingHandler.changeTrackingLastVersion)
+      jdbcContext._changeTrackingLastVersion = () =>
+        Some(changeTrackingHandler.changeTrackingLastVersion())
     }
 
-    var jdbcService: JdbcService = new JdbcService(jdbcContext)
-
-    if (config.splitBy.nonEmpty) {
-      jdbcService = new JdbcPartitionedService(jdbcContext)
-    }
-
-    if (config.isChangeTrackingEnabled && config.CTChangesQuery.nonEmpty) {
-      jdbcService = new JdbcCTService(jdbcContext)
+    val jdbcDF: DataFrame = if (config.CTChangesQuery.isEmpty) {
```

Comment:
in this case, normal CT won't ever work, is that so?

Reply:
If `config.CTChangesQuery.isEmpty` then `JdbcService(jdbcContext)` is instantiated and `changeTrackingHandler.query` is passed (as `query` from `jdbcContext`) to its method `loadData`. If that's what you mean. Works fine.